eratos_xarray/backend/eratos_.py

```diff
@@ -103,8 +111,20 @@ def __init__(self, gsdata):
         self.gsdata = gsdata
 
     @classmethod
-    def open(cls, ern, eratos_auth: AccessTokenCreds = None):
-        adapter = Adapter(eratos_auth)
+    def open(
+        cls,
+        ern,
+        eratos_auth: Optional[BaseCreds] = None,
```

Comment:
I think the main point is that the following line is misleading at best since a valid `BaseCreds` class isn't a valid authentication method.
```suggestion
        eratos_auth: Optional[Union[AccessTokenCreds, AccessBearerCreds, JobKeyCreds]] = None,
```

Reply:
I think the usual pattern is to use the parent class in type hinting - I'm not sure if it's necessarily misleading since in a lot of SDK docs and examples `AccessTokenCreds` are used and the name `BaseCreds` implies that it's a base class. This should be clear with documentation.

Reply:
I think its a bit misleading since the `eratos.adapter.Adapter` class states that `BaseCreds` is the type for the `creds` param. The hatchet is kinda burried a little bit but you would be right if `BaseCreds.auth_query_params` didn't throw an error. Since it does it invalidates the contract its setting. 

If you think of the user experience (and our current lack of documentation) here its potentially setting up users for failure since using this `BaseCreds` class for auth will just result in `NotImplementedError` when attempting to get an auth token from the tracker which is a poor user experience.

If we can fixup `BaseCreds` with the following you'd be correct in using it... But I'd argue the user experience is lacking since people in practice use type hints to indicate the literal types expected.

```python
from abc import ABC, abstractmethod
...
class BaseCreds(ABC):
    """
    A base credentials class to interact with the tracker node.
    """
...
    @abstractmethod
    def auth_query_params(self):
        """
        Returns authentication query parameters.
        
        Returns
        -------
        dict
            A dictionary containing authentication parameters.
        """
        pass
```